README.md

```diff
@@ -2,31 +2,201 @@
+--secondNumber;
+```
+
+When we place the operator before the variable, it will still have the same
+result as before: `++firstNumber` will return `21` and `--secondNumber` will
+return `9`. But order still matters! Let's take a look at an example of when
+the placement of the increment or decrement operator would make a difference:
+
+```java
+int result = ++firstNumber;
+System.out.println(result);
+
+// Reset firstNumber to 20
+firstNumber = 20;
+result = firstNumber++;
+System.out.println(result);
+```
+
+Now let's look at the output of the code above:
+
+```text
+21
+20
+```
+
+So why are the outputs different? When the first line is run with the `++`
+preceding the variable, Java will first perform the operation of
+`firstNumber = firstNumber + 1;` before assigning the value of `firstNumber`
+to `result`. The next time when the increment operator is used is in the
+format of `firstNumber++`. When the `++` follows the variable, Java will
+access the value of `firstNumber` before completing the increment. This is
+because:
+
+```java
+result = firstNumber++;
+```
+
+is equivalent to
+
+```java
+result = firstNumber;
+firstNumber = firstNumber + 1;
+```
+
+Therefore, `result` will be assigned the value 20 before `firstNumber` is
+incremented.
+
+To avoid these types of issues, it is best to not combine the assignment
+operator, `=`, with the increment or decrement operators in the same statement.
+
+Now let's discuss the other compound assignment operators:
+
+- `+=` is the addition compound assignment operator. It is a binary operator
+  where the value of the expression on the right-hand side will be added to the
+  variable on the left-hand side of the operator and then stored in that
+  variable. Consider the example:
+  - `firstNumber += secondNumber;` is the same as
+    `firstNumber = firstNumber + secondNumber;`
+  - In the example above, the addition compound operator will first add the
+    value of `secondNumber` to the value stored in `firstNumber` and then
+    re-assign the result of that expression to `firstNumber`. So after the
+    statement has been executed,`firstNumber` will have a value of 30.
+  - The addition compound assignment operator is an abbreviated statement of
+    variable = variable + right-hand expression.
+  - To see this played out in memory, consider the following:
+
+    <iframe width="800" height="500" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=public%20class%20Main%20%7B%0A%20%20%20%20public%20static%20void%20main%28String%5B%5D%20args%29%20%7B%0A%20%20%20%20%20%20int%20firstNumber%20%3D%2020%3B%0A%20%20%20%20%20%20int%20secondNumber%20%3D%2010%3B%0A%20%20%20%20%20%20%0A%20%20%20%20%20%20firstNumber%20%2B%3D%20secondNumber%3B%0A%20%20%20%20%20%20System.out.println%28firstNumber%29%3B%0A%20%20%20%20%7D%0A%7D&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=3&heapPrimitives=nevernest&origin=opt-frontend.js&py=java&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
```

Comment:
On GitHub, I'm just seeing the HTML here.

Reply:
Linda claims this will work once pushed to Canvas.

Reply:
Yes, look at https://learning.flatironschool.com/courses/6676/pages/embedding-code-visualization-in-a-lesson

It takes a few seconds to load.